src/pmfs/vos_tasks.h

```diff
@@ -0,0 +1,73 @@
+/**
+ * (C) Copyright 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#ifndef VOS_TASKS_H
+#define VOS_TASKS_H
+
+#include <gurt/types.h>
+#include <daos/object.h>
+#include <daos_types.h>
+#include <daos_obj.h>
+
+#include <spdk/env.h>
+
+typedef int (*task_comp_cb_t)(void *cb_args, int rc);
+
+enum task_op {
+	VOS_OBJ_UPDATE = 1,
+	VOS_OBJ_FETCH,
+	VOS_OBJ_PUNCH,
+	VOS_OBJ_GET_NUM_DKEYS,
+	VOS_OBJ_LIST_DKEYS,
+};
+
+struct vos_client_obj_list_args {
+	daos_handle_t coh;
+	daos_unit_oid_t oid;
+	/* number of KEYs */
+	uint32_t *nr;
+	/* total buffer length of KEYs */
+	uint64_t *len;
+	/* Key descriptor used for key enumeration */
+	daos_key_desc_t	*kds;
+	/* data buffer to store KEYs */
+	void *buf;
+};
+
+struct vos_client_obj_rw_args {
+	daos_handle_t coh;
+	daos_unit_oid_t oid;
+	daos_epoch_t epoch;
+	uint32_t pm_ver;
```

Comment:
general comment, you're mixing 32-bit and 64-bit fields so this structure is less space efficient than it could be.  Better to group 32-bit fields.

Reply:
ACK